src/main/scala/com/tuplejump/snackfs/api/model/MakeDirectoryCommand.scala

```diff
@@ -18,39 +18,44 @@
  */
 package com.tuplejump.snackfs.api.model
 
-import org.apache.hadoop.fs.Path
-import org.apache.hadoop.fs.permission.FsPermission
-import scala.util.{Failure, Success, Try}
 import scala.concurrent.Await
-import com.tuplejump.snackfs.fs.model.{FileType, INode}
-import com.twitter.logging.Logger
 import scala.concurrent.duration.FiniteDuration
-import com.tuplejump.snackfs.cassandra.partial.FileSystemStore
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.fs.permission.FsPermission
 import com.tuplejump.snackfs.api.partial.Command
+import com.tuplejump.snackfs.cassandra.partial.FileSystemStore
+import com.tuplejump.snackfs.fs.model.FileType
+import com.tuplejump.snackfs.fs.model.INode
+import com.tuplejump.snackfs.util.LogConfiguration
+import com.twitter.logging.Logger
+import com.tuplejump.snackfs.security.UnixGroupsMapping
+import org.apache.hadoop.fs.permission.FsAction
 
 object MakeDirectoryCommand extends Command {
   private lazy val log = Logger.get(getClass)
 
-  private def mkdir(store: FileSystemStore,
-                    filePath: Path,
-                    filePermission: FsPermission,
-                    atMost: FiniteDuration): Boolean = {
-
+  private def mkdir(store: FileSystemStore, filePath: Path,  filePermission: FsPermission, atMost: FiniteDuration): Boolean = {
```

Comment:
Please remove the formatting changes... We can have them as separate commit later if necessary. In this commit we should only focus on changes in code and design.

I will actually recommend we split the pull requests into smaller pieces focusing on specific changes.